### PR TITLE
bump image tags in the CSI chart to the latest ones in the upstream

### PR DESCRIPTION
**Pull request by jiaqiluo**
#### Pull Request Checklist ####

- [ ] Any new images or tags consumed by charts has been added [here](https://github.com/rancher/image-mirror)
- [ ] That helm lint and pack run successfully on the chart.
- [ ] Deployment of the chart has been tested and verified that it functions as expected.
- [ ] Changes to scripting or CI config have been tested to the best of your ability

#### Types of Change ####

version bump + chart info update

Note: 
Bumping image tags is the only change in the upstream CSI chart 
between 2.6.1 and 2.6.0, 
between 2.5.3 and 2.5.2, 
between 2.4.3 and 2.4.2  

#### Linked Issues ####
https://github.com/rancher/rancher/issues/38188
https://github.com/rancher/rke2/issues/3377

#### After the PR is merged ####

Once the PR is merged, typically upon a new release, please post an announcement in the following channels:

* #discuss-rancher-feature-charts
* #discuss-rancher-feature-vsphere
* #discuss-rancher-k3s-rke2

### CODE REVIEW

**Review by brandond — CHANGES_REQUESTED** [review]
I think we should also bump the version and appVersion to 2.6.1? I think that might be why CI is failing too.

https://github.com/rancher/vsphere-charts/blob/21dfa026688d77a9aff256ffdea0bf5a4af196c1/charts/rancher-vsphere-csi/Chart.yaml#L11

**Comment by brandond** [comment]
Are we not bumping the chart version to 1.6.1 to track the images?

**Review by brandond — APPROVED** [review]

File: charts/rancher-vsphere-csi/Chart.yaml [hunk 1]
```diff
@@ -1,22 +1,22 @@
 annotations:
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: vSphere CSI
-  catalog.cattle.io/kube-version: '>= 1.20.0-0 < 1.24.0-0'
+  catalog.cattle.io/kube-version: '>= 1.20.0-0 < 1.26.0-0'
   catalog.cattle.io/namespace: kube-system
   catalog.cattle.io/os: linux,windows
   catalog.cattle.io/permits-os: linux,windows
   catalog.cattle.io/rancher-version: '>= 2.6.0-0 < 2.7.0-0'
```
Comment (jakefhyde) [inline charts/rancher-vsphere-csi/Chart.yaml:8]: Does this need to be updated for `v2.7.0`?

Comment (jiaqiluo) [reply charts/rancher-vsphere-csi/Chart.yaml]: Although it does not affect RKE2 to build its own chart based on this, I decided to remove the upper bound of the rancher version now so we do not need to update the chart again when we bring it into rancher/chart later.

**Review by jakefhyde — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Review by snasovich — APPROVED** [review]

**Pull request merged**